gpflow/decors.py

```diff
@@ -123,14 +123,16 @@ def compute_something(self):  # self is parameterized object.
         return s + b
     ```
 
+    :param objs: Node (subclass) instances
     :param convert: Flag which is used for turning tensor convertion
         feature on, `True`, or turning it off, `False`.
     """
-    prev_value = _params_as_tensors_enter(obj, convert)
+    prev_values = [_params_as_tensors_enter(o, convert) for o in objs]
     try:
         yield
     finally:
-        _params_as_tensors_exit(obj, prev_value)
+        for o, pv in zip(objs, prev_values):
+            _params_as_tensors_exit(o, pv)
```

Comment:
I think the changes in this file would best have been done in a separate PR. I think it's best if @awav signs off on these. I'm sure they're fine. I just don't know the code inside out.

Reply:
No problem, was thinking that as well (after the fact), but I can still make a PR just for that commit.

Reply:
See #672